### PR TITLE
remove List::MoreUtils dep

### PR DESCRIPTION
**Pull request by haarg**
This drops the dep on List::MoreUtils by either switching to List::Util functions where appropriate, or inlining things otherwise.  This is an alternative to #118.

### CODE REVIEW

**Comment by autarch** [comment]
This all looks good to me. I'm fine with not using List::SomeUtils as long as we don't care about whatever speed optimizations its XS implementation might offer (which might be none).

File: dist.ini [hunk 1]
```diff
@@ -216,8 +216,8 @@ skip = ^Moose::Meta::TypeConstraint::Union$
 ;authordep Eval::Closure               = 0.04
            List::MoreUtils             = 0.28
 ;authordep List::MoreUtils             = 0.28
-           List::Util                  = 1.35
-;authordep List::Util                  = 1.35
+           List::Util                  = 1.44
+;authordep List::Util                  = 1.44
```
Comment (karenetheridge) [inline dist.ini]: These should be 1.45, when it's ready -- to make sure we keep the same undef vs '' behaviour.

**Comment by karenetheridge** [comment]
The speed change is probably minimal, but would be good to bench it anyway just in case.

**Comment by karenetheridge** [comment]
Thanks, merged!  Will release as 2.1700-TRIAL (release window to go stable is the end of April - I've marked my calendar)

**Comment by karenetheridge** [comment]
There's a few mentions of LMU::zip in pod that we could possibly replace with something else -- one of the pair* subs in List::Util might do the job there.

**Pull request closed without merging**